ESD109-Shadowsedit/ShadowsOfTheKnight_all655.cpp

```diff
@@ -0,0 +1,183 @@
+// block and ran out of time as I was not sure yet how I would configure each
+// if-statement. Normally I would keep trying and experimenting until something
+// concrete came to mind. At this point, which was about 15 minutes, I made some
+// notes/drawings, which are shown at the end of this document.
+//
+// 
+//
+// When I wrote out my ideas on paper as instructed in step 4 (see
+// [Coding with Reflection assignment instructions](https://canvas.msstate.edu/courses/133359/assignments/1257462)),
+// it became clear what I needed to do as well as helped me realize scenarios
+// that I had not considered beforehand. I essentially solved the problem by
+// using a box or four separate variables as boundaries, which encompassed the
+// points that the bomb could be in, and points outside of the box’s boundaries
+// were eliminated from the realm of possible points. For example, if the
+// direction of the bomb is up-left (UL) of Batman, then I know that all points
+// below Batman (DL, D, DR) and above as well as to the right of Batman (U, UR)
+// are out of the realm of points to consider, so in this case, the bottom right
+// corner of the box would be where Batman is currently positioned when informed
+// that the bomb is UL. If the bomb was located DR (Down-Right) of Batman,
+// Batman’s current position would be the top-left of the box (see **Figure
+// 1**). I basically created a binary search but for **two** dimensions without
+// using an array. Using an array might have been much cleaner and concise but
+// doing it this way made more sense for me. In my code, I kept the box’s
+// position updated, so it would essentially shrink any which way with each jump
+// that Batman made. Whenever Batman jumped, he would jump to the center of the
+// box every time until he discovered the bomb. Please see the code for more
+// details as well as the following figures/notes.
+//
+// <img style="border-style: ridge;" title="fig1&amp;fig2.png" src="fig1&amp;fig2.png" alt="" width="700" height="275">
+//
+// ### Here are some notes/drawings I made:
+//
+// <img style="border-style: ridge;" title="notes1.jpg" src="notes1.jpg" alt="" width="700" height="1200">
+// <img style="border-style: ridge;" title="notes2.jpg" src="notes2.jpg" alt="" width="700" height="1200">
+// <img style="border-style: ridge;" title="notes3.jpg" src="notes3.jpg" alt="" width="700" height="700">
+//
+// 
+// In this sketch, I realized what the math would look like or involve to find
+// the center of the “box”. This was the moment that I grasped it was just some
+// simple algebra.
+//
+// 
+//
+// ## The code is as follows:
+#include <iostream>
+#include <string>
+#include <vector>
+#include <algorithm>
+
+using namespace std;
+
+/**
+ * Auto-generated code below aims at helping you parse
+ * the standard input according to the problem statement.
+ **/
+
+int main()
+{
+    int w; // width of the building.
```

Comment:
Put comments on the preceeding line, so the CodeChat Editor will display them as a doc block.